compiler/rustc_codegen_llvm/src/llvm_util.rs

```diff
@@ -321,6 +321,10 @@ pub fn target_features(sess: &Session, allow_unstable: bool) -> Vec<Symbol> {
             }
         })
         .filter(|feature| {
+            // skip checking special features, as LLVM may not understands them
+            if RUSTC_SPECIAL_FEATURES.contains(feature) {
+                return true;
+            }
```

Comment:
This makes it so that `cfg!(target_feature = "backchain")` will always be true, even if it is not enabled. Is that the intended behavior? It seems very odd.

Reply:
No, this is to skip checking if LLVM supports this attribute. The logic for adding this attribute is somewhere else.
If you are unconvinced, check out this Godbolt playground: https://godbolt.org/z/77hrrsbo1.

Reply:
I am convinced that the logic is wrong, see https://github.com/rust-lang/rust/issues/129927.

Reply:
Also, I have no idea what I am supposed to look at in this Godbolt.^^ I know nothing about s390x or `backchain`, I am just noticing odd things in the rustc source code.